### DISCUSSION TITLE
PDF : number-sections: true conflicts with cleverref and adds empty space to acronyms

### DISCUSSION THREAD

**Discussion by Minh-AnhHuynh** [quarto-dev/quarto-cli#11390] [Q&A]
### Description

It seems that since some recent [TeX update](https://tex.stackexchange.com/questions/730126/update-to-cleveref-firstaid-adds-space-after-tcolorbox-documentation-commands), there is a conflict between the way cleveref handles hyperlink. In my case, it conflicts with the acronym package and the `number-sections: true` parameter in the Quarto yaml. The problem occurs when an acronym is generated, there is a blank space introduced the first time there is an acronym defined.

Is there some kind of alternative or latex command I could use to fix my issue ?
I need `number-sections: true` in my document.
The reason I use cleveref is because it allows the reference to be hyperlinked and not just the number. I also defined commands that automatically bold the reference:
Else I would have to drop cleveref and rewrite my entire document, which is not practical.

```yaml
header-includes:
- \crefdefaultlabelformat{#2\textbf{#1}#3} # <-- Only #1 in \textbf
- \crefformat{subfigure}{#2\textbf{#1}#3}
- \crefname{table}{\textbf{table}}{\textbf{tables}}
- \Crefname{table}{\textbf{Table}}{\textbf{Tables}}
- \crefname{figure}{\textbf{figure}}{\textbf{figures}} 
- \Crefname{figure}{\textbf{Figure}}{\textbf{Figures}}
- \crefname{subfigure}{\textbf{figure}}{\textbf{figures}}
- \Crefname{subfigure}{\textbf{Figure}}{\textbf{Figures}}
```

Reproductible example:

````qmd
---
format:
  pdf:
    pdf-engine: lualatex
    papersize: a4
    number-sections: true
header-includes:
- \usepackage{suffix}
- \usepackage{acronym} # Needs suffix and e-TeX
- \usepackage{hyperref}
- \usepackage[capitalise, noabbrev, nameinlink]{cleveref}
---

\begin{acronym}
  \acro{VDR}{Vitamin D Receptor}
  \acro{VDRE}{Vitamin D Receptor Element}
\end{acronym}

# Test acronyms

Problem here : 

- Text \acs{VDR}
- Unique usage (\acsu{VDR})

- Unique usage (\acs{VDRE})
- Acronym long here \acl{VDRE}
- First time definition of \ac{VDRE}

No problem :

- Second usage (\ac{VDR})
- Text \acl{VDR}
- Acronym long here \acl{VDRE}
- Text \ac{VDRE}
````

**Comment by mcanouil** [comment]
What version of Quarto are you using?
Did you try to look at the intermediate LaTeX file to debug it? `keep-tex: true`.

To manage expectations, it's unlikely Quarto will make substantial changes to Pandoc template.
As a reminder, you the templates/partials, if you manage to fix the conflict: https://quarto.org/docs/journals/templates.html#latex-partials

**Comment by Minh-AnhHuynh** [comment]
I am using quarto version 1.6.33.

I have looked at the .tex file but I am not familiar enough with latex to know how to debug this. Besides, there is no mention of a specific cleveref or acronym code I could debug. Moreover, now that I'm looking more intently at two files generated before and after the TinyTeX update of 2024.11, the generated TeX files are identical.
[cleveref-conflict-file.txt](https://github.com/user-attachments/files/17736171/cleveref-conflict-file.txt).

For now the easiest fix for me should be to reinstall TinyTeX and switch version using `tinytex::install_tinytex(version = "2024.10")`. However it means my document will not be compatible with future TinyTeX versions which is not ideal.

Initially I thought quarto would be the problem but it seems the problem comes from the TeX version update of November 2024.

**Reply by mcanouil** [reply]
> Initially I thought quarto would be the problem but it seems the problem comes from the TeX version update of November 2024.

Unfortunately, if the LaTeX distribution update broke your document, there is not much if not nothing Quarto can do.
Working with LaTeX packages is a source of headaches on its own.
Quarto cannot really offer support for any custom things users might want to do especially because at this point, it means the users are supposed to learn the underlying technology (LaTeX, Typst, OpenXML, HTML, CSS).


You could try asking on "LaTeX forums" about your issue, but it would require of you to strip components of the LaTeX file to end up with something small and manageable.
